### DISCUSSION TITLE
validation hooks

### DISCUSSION THREAD

**Discussion by sebastiangnw** [pocketbase/pocketbase#5688] [Q&A]
http://127.0.0.1:8090/api/collections/users/records

{
    "username": "test",
    "email": "test@gmail.com",
    "emailVisibility": true,
    "password": "12345678",
    "passwordConfirm": "12345678",
    "name": "test",
    "last_name": "test"
}

Hello, how can i get de password and passwordConfirm to make validations?

``

onRecordBeforeCreateRequest((e) => {
    console.log("Hook - antes de crear usuario");

    console.log("Registro completo:", e.password);
    
    // Convertir el registro a JSON y asignarlo a una variable
    const registro = JSON.stringify(e.record);
    console.log("Registro completo:", registro); // Imprimir registro completo

    // Parsear el registro para acceder a los valores como un objeto
    const registroObjeto = JSON.parse(registro);

    // Extraer el email, password y passwordConfirm
    const email = registroObjeto.email || ""; // Asegúrate de que se maneje como una cadena
    const password = registroObjeto.password || "";
    const passwordConfirm = registroObjeto.passwordConfirm || "";

    console.log("Email:", email)
    console.log("Password:, password")
    console.log("PasswordConfirm:", passwordConfirm)

    // Validar el campo email
    if (!email) {
        throw new ApiError(400, "Errores de validación", {
            "email": new ValidationError("required", "El campo email es obligatorio."),
        });
    }

    // Validar el campo password
    if (!password) {
        throw new ApiError(400, "Errores de validación", {
            "password": new ValidationError("required", "El campo password es obligatorio."),
        });
    }

}, "users");

``

**Comment by hungcrush** [comment]
why you have to do that ?

**Reply by sebastiangnw** [reply]
I want to handle errors in another language because the current errors are in English. Additionally, I would like to return a customized response when a request is made to the endpoint.

**Reply by hungcrush** [reply]
I think `routerAdd(...)` might be the solution you're looking for. In this scenario, you would create your own register API and use `form.loadRequest` to create the user.

refer: https://pocketbase.io/docs/js-records/#update-record-with-data-validations
